core/src/main/scala/formula/Form.scala

```diff
@@ -140,10 +139,18 @@ object Form {
             acc.zip(v.get).map { case (acc, v) => acc.appended(v) }
           }
 
-        override def set(a: List[A]): Unit =
-          a.zip(variables.now()).foreach { case (a, v) =>
-            v.set(a)
-          }
+        override def set(a: List[A]): Unit = {
+          println(s"set length to ${a.length}")
+          countForm.set(a.length)
+          setTimeout(
+            () =>
+              a.zip(variables.now()).foreach { case (a, v) =>
+                println(s"set variable to $a")
+                v.set(a)
+              },
+            50
```

Comment:
Not very happy about the solution here, pls let me know if you have suggestions on how to improve this.